### ISSUE TITLE
Mate weather applet fails to get weather

### ISSUE THREAD

**Issue by duairc** [NixOS/nixpkgs#53700]
## Issue description

The weather applet included in mate-applets fails to download weather information. It just shows "0°F", or sometimes "--" instead. I've tried setting it to different cities all over the world, none of them work.

Interestingly, for UK cities, I get see the following in `journalctl -xe`, perhaps it fails for a similar reason on other cities (though I don't see an error message for these):

```
.mateweather-ap[NNNN]: Failed to get Met Office forecast data: 6 TLS/SSL support not available; install glib-networking.
```

### Steps to reproduce

Install the weather applet in MATE and try to get it to show anything useful.

I've tried adding `glib-networking` to the `buildInputs` of both `mate-applets` and `libmateweather`, but it doesn't help at all.

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.13, NixOS, 19.03pre165037.eebd1a92637 (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - channels(shane): `""`
 - channels(root): `"nixos-19.03pre165037.eebd1a92637"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by jtojnar** [comment]
Adding `glib-networking` to `libmateweather` would not do anything, since `glib-networking` is loaded from the runtime environment. `wrapGAppsHook` will wrap executables in `bin` and `libexec` directories, setting `GIO_EXTRA_MODULES` variable correctly – that is an operation which does not make sense for libraries, it should work for `mate-applets`, though.

**Comment by duairc** [comment]
...it works now. The only thing that's changed is that I now have Gnome and MATE installed side-by-side, whereas before I only had MATE. It seems a lot of things in MATE need bits of Gnome to work properly. I can't say for sure that's what was causing this, but I just can't think of anything else it could have been. In any case I'm going to close this.

**Comment by AshleyYakeley** [comment]
Ran into this, please reopen. I'm running MATE on NixOS, but not Gnome.

**Comment by HammyHavoc** [comment]
Yep, also seeing this: https://github.com/NixOS/nixpkgs/issues/311066

**Comment by bobby285271** [comment]
2 ugly workarounds are done right now:

- #262496 propagated `glib-networking`
- #311074 enabled `ervices.gnome.glib-networking`

So I assume this is already fixed in 23.11. Note that since https://github.com/NixOS/nixpkgs/pull/289062 the weather applet is built in-process since we want to package the wayland session.
